### ISSUE TITLE
Curious color behavior

### ISSUE THREAD

**Issue by N247S** [hananils/kirby-colors#12]
Blueprint setup
```yml
  header_background_color:
    label: Header color
    type: colors
    alpha: true
    width: 1/2
```

Php context
```php
# used this method to obtain the color-string
$site->header_background_color()->toColor();
```

input:
![image](https://user-images.githubusercontent.com/10078398/97054387-fda13900-1584-11eb-8644-6a19a1cebf81.png)

output:
![image](https://user-images.githubusercontent.com/10078398/97054604-5cff4900-1585-11eb-8f8e-d5db50cf8358.png)

If more information is required, let me know!

**Comment by N247S** [comment]
Input:
![image](https://user-images.githubusercontent.com/10078398/97055156-866ca480-1586-11eb-8956-24d623a77e17.png)

Output:
![image](https://user-images.githubusercontent.com/10078398/97055199-9f755580-1586-11eb-979c-596276c4413f.png)

When alpha is included, even weirder things happen

**Comment by nilshoerrmann** [comment]
Thanks for you report! 
Which values get stored in the content file in both cases?

**Comment by N247S** [comment]
I am sorry, but I have stepped over to another color-plugin. So I don't have a/the test-setup anymore. From what I could test is that it was consistent when I entered the exact numbers above. So maybe you can test it for yourself? (from what I could gather: it only happens when there is one or more `0x` number in the rgb(a)-string. Maybe that gives enough information?

**Comment by nilshoerrmann** [comment]
I just checked this out and cannot reproduce on either `master` or `develop` branch. 
I'll leave the issue open in case someone else is able to provide more information.

**Comment by N247S** [comment]
Got some spare time, so I reïnstalled the setup (I installed it by downloading the zip from this repository btw.)
The result that is saved is correct: `#01980b` and `#01930bcc`, but apparently the `toColor` function doesn't work properly.

Also during testing I found another curious thing. I first tested with alpha (so an `rgba` value was stored). When I disable the alpha, the display doesn't show the alpha indicator anymore, but the previously stored alpha is retained (visually in the preview-display; and after editing the field / resaving the form). When I then requested the color-string using `toColor`, it still included the alpha, eventhough it was not specified! (Either this is a feature, or a bug. Whatever the usecase may be 😛 ).

Could be an interesting feature though, 'locking' an attribute in the color-picker.

**Comment by squareclouds** [comment]
hi @nilshoerrmann ,

i can confirm this. i am having trouble as well. the hex color is saved correctly, but when i output it in php it comes out wrong. if i change from hex to rgb in the dropdown of the field, the php output is correct.

value i am having trouble with:
0b6a23

it comes out as:
b06a23

so the first two digits are switched. as you can see, @N247S had also a problem with a hex value starting with zero.

so:

until now i can also confirm that it might be a problem with the initial zero. if i change it to 1 everything works fine, and if i leave it as 0 and the second digit changes to any other hex value the switching happens again.

so regarding @N247S 's problem, it has nothing to do with the alpha, since i have alpha deactivated in my field.

can you please look into this?

**Comment by squareclouds** [comment]
PS: this should be technically marked as a bug :) @nilshoerrmann did you run your tests with the same color as @N247S ?

**Comment by nilshoerrmann** [comment]
Thanks @squareclouds for your comments. I'll see if I can reproduce this now.

> @nilshoerrmann did you run your tests with the same color as @N247S ?

Yes, I did, that's the strange part of this issue.

**Comment by squareclouds** [comment]
hmm strange. just FYI these are the plugins i am using, in case there might be some collision between them, my panel language is in english (just in case) and i am using kirby 3.5.4

![grafik](https://user-images.githubusercontent.com/9073054/118259206-294c2800-b4b1-11eb-9fdc-ba09e8c1110d.png)

**Comment by nilshoerrmann** [comment]
Which PHP version are you using?

**Comment by squareclouds** [comment]
7.4, let me switch quickly to 8

**Comment by nilshoerrmann** [comment]
Would you mind checking if changing this

https://github.com/hananils/kirby-colors/blob/4153597732cd9ff0f222f58d0a2822aecb0a6a13/lib/Converter.php#L140-L142

to 

```php
if ($pad) {
    $hex = str_pad($hex, 2, '0', STR_PAD_LEFT);
}
```

fixes your problem?

**Comment by squareclouds** [comment]
ok i changed to 8 and phpinfoed to be sure it was active  already (it was) and it still happens

**Comment by nilshoerrmann** [comment]
@squareclouds: Was that with or without the suggested change?

**Comment by squareclouds** [comment]
yes it does! :) all good now. everything else still looks fine but again, i am not using alpha or anything else.

**Comment by squareclouds** [comment]
sorry that was before, with the code change it works fine

**Comment by nilshoerrmann** [comment]
That's great! I'll prepare a new version with that change.

**Comment by nilshoerrmann** [comment]
This is now fixed in version 1.3.0:
https://github.com/hananils/kirby-colors/releases/tag/1.3.0
